src/factory/mod.rs

```diff
@@ -334,15 +370,23 @@ impl Factory {
 
     /// Create a `Mesh` sharing the geometry with another one but with a different material.
     /// Rendering a sequence of meshes with the same geometry is faster.
-    pub fn mesh_instance_with_material<M: Into<Material>>(
+    pub fn duplicate_mesh_with_material<M: Into<Material>>(
         &mut self,
         template: &Mesh,
         material: M,
     ) -> Mesh {
         let mut hub = self.hub.lock().unwrap();
+        let instances = self.backend
```

Comment:
might want to have a helper private method in the factory for this, given how many times it's repeated